diff --git a/node_modules/playactor/dist/discovery/udp.js b/node_modules/playactor/dist/discovery/udp.js
index ea6e8bc..bd24770 100644
--- a/node_modules/playactor/dist/discovery/udp.js
+++ b/node_modules/playactor/dist/discovery/udp.js
@@ -56,15 +56,19 @@ class UdpSocketManager {
     }
 }
 class UdpDiscoveryNetwork {
-    constructor(socketManager, boundPort, socket, port, version) {
+    constructor(socketManager, boundPort, socket, port, version, onMessage) {
         this.socketManager = socketManager;
         this.boundPort = boundPort;
         this.socket = socket;
         this.port = port;
         this.version = version;
+        this.onMessage = onMessage;
     }
     close() {
         debug("closing udp network");
+        if (this.onMessage) {
+            this.socket.removeListener("message", this.onMessage);
+        }
         this.socketManager.release(this.boundPort);
     }
     ping(deviceIp) {
@@ -142,7 +146,7 @@ class UdpDiscoveryNetworkFactory {
         else {
             debug("joining existing socket for ", config);
         }
-        return new UdpDiscoveryNetwork(this.socketManager, bindPort, socket, this.port, this.version);
+        return new UdpDiscoveryNetwork(this.socketManager, bindPort, socket, this.port, this.version, onMessage);
     }
     createDevices(config, onDevice) {
         return this.createMessages(config, message => {
